### ISSUE TITLE
Marco in action

### ISSUE THREAD

**Issue by grasswang** [tynanford/phoebusgen#4]
Hi, Is it possible to add macro to "_Action"? 

> 
       BPMB='PREFIX:'
        actionB = w.ActionButton(B_name,B_name, B_pv_name, x_position+widget_length, y_position, widget_length, widget_height)
        actionB._prop_factory.add_macro('BPM', BPMB)
        actionB.action_open_display('main_bpm.bob','tab')

and I end up getting the macro outside of action so the macro didn't pass to the new display

>        
        <macros>
          <BPM>PREFIX:</BPM>
        </macros>
        <actions>
          <action type="open_display">
            <description>Open Display</description>
            <file>main_bpm.bob</file>
            <target>tab</target>
          </action>
        </actions>

Thanks!

**Comment by tynanford** [comment]
Hi, yes that is a good idea. I'll work on adding a .macro method to ActionButton class.  Thanks!

**Comment by tynanford** [comment]
This should be fixed in 9c36c91  

For now, actions are simply methods. So I added the macros option to the method. Eventually, it might make sense for an Action to be a class with its own methods so you don't have to define everything in the parameter list.

 ```
>>> from phoebusgen import widget as w
>>> a = w.ActionButton('test', 'test', 'test',0,0,0,0)
>>> a.action_open_display('main_bpm.bob','tab', macros={'BPM': 'BPMA'})
>>> a
<?xml version="1.0" ?>
<widget type="action_button" version="2.0.0">
  <name>test</name>
  <x>0</x>
  <y>0</y>
  <width>0</width>
  <height>0</height>
  <pv_name>test</pv_name>
  <text>test</text>
  <actions>
    <action type="open_display">
      <description>Open Display</description>
      <macros>
        <BPM>BPMA</BPM>
      </macros>
      <file>main_bpm.bob</file>
      <target>tab</target>
    </action>
  </actions>
</widget>
```

Uploaded to pip so you can upgrade the package

**Comment by grasswang** [comment]
Wow, it was fast, and it works! Thanks a lot!
